pkg/webhook/policy_mutating.go

```diff
@@ -0,0 +1,71 @@
+package webhook
+
+import (
+	"context"
+
+	opa "github.com/open-policy-agent/frameworks/constraint/pkg/client"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission/builder"
+	atypes "sigs.k8s.io/controller-runtime/pkg/webhook/admission/types"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	CreateWebhookFuncs = append(CreateWebhookFuncs, AddMutatingWebhook)
+}
+
+// AddMutatingWebhook registers the mutating webhook server with the manager
+// below: notations add permissions kube-mgmt needs. Access cannot yet be restricted on a namespace-level granularity
+// +kubebuilder:rbac:groups=*,resources=*,verbs=get;list;watch
+// +kubebuilder:rbac:groups=,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+func AddMutatingWebhook(mgr manager.Manager, opa opa.Client) (webhook.Webhook, error) {
+	mutatingWH, err := builder.NewWebhookBuilder().
+		Mutating().
+		Name(mutatingWHName).
+		Path("/v1/mutate").
+		Rules(admissionregistrationv1beta1.RuleWithOperations{
+			Operations: []admissionregistrationv1beta1.OperationType{admissionregistrationv1beta1.Create, admissionregistrationv1beta1.Update},
+			Rule: admissionregistrationv1beta1.Rule{
+				APIGroups:   []string{"*"},
+				APIVersions: []string{"*"},
+				Resources:   []string{"*"},
+			},
+		}).
+		Handlers(&mutationHandler{opa: opa, client: mgr.GetClient()}).
+		WithManager(mgr).
+	//	ForType(&corev1.Pod{}). // https://github.com/kubernetes-sigs/controller-runtime/pull/81/files
+		Build()
+	if err != nil {
+		return mutatingWH, err
+	}
+
+	return mutatingWH, nil
+}
+
+var _ admission.Handler = &mutationHandler{}
+
+type mutationHandler struct {
+	opa     opa.Client
+	client  client.Client // change this later to ctrl or something similar
```

Comment:
client.Client is a client of the k8s API server, for context.

Reply:
Would it make sense to change it to something like `k8s` to make its use more evident?

Reply:
k8s works for me. I think there is some amount of personal preference.